tss-esapi/src/structures/capability_data.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use crate::{
+    constants::CapabilityType,
+    structures::{
+        AlgorithmPropertyList, CommandCodeAttributesList, CommandCodeList, EccCurveList,
+        HandleList, PcrSelectionList, TaggedPcrPropertyList, TaggedTpmPropertyList,
+    },
+    tss2_esys::{TPM2_CAP, TPM2_MAX_CAP_BUFFER, TPMS_CAPABILITY_DATA, TPMU_CAPABILITIES},
+    Error, Result, WrapperErrorKind,
+};
+use log::error;
+use std::convert::{TryFrom, TryInto};
+use std::mem::size_of;
+
+/// A representation of all the capabilites that can be associated
+/// with a TPM.
+///
+/// # Details
+/// This corresponds to `TPMS_CAPABILITY_DATA`
+#[derive(Debug, Clone)]
+pub enum CapabilityData {
+    Algorithms(AlgorithmPropertyList),
+    Handles(HandleList),
+    Commands(CommandCodeAttributesList),
+    PpCommands(CommandCodeList),
+    AuditCommands(CommandCodeList),
+    AssignedPcr(PcrSelectionList),
+    TpmProperties(TaggedTpmPropertyList),
+    PcrProperties(TaggedPcrPropertyList),
+    EccCurves(EccCurveList),
+    // These are in the TPM TPMU_CAPABILITIES, but are not defined by esapi-2.4.1
+    // AuthPolicies(),
+    // ActData(),
+}
```

Comment:
:thinking: Hmm... I wonder if we need [`#[non_exhaustive]` on the enum](https://turreta.com/2019/12/20/using-non_exhaustive-for-non-exhaustive-enums/) if this could be further extended with AuthPolicies and ActData... Adding more variants later would be a breaking API change without this attribute I think.

Reply:
Good spot!

Reply:
I wrote a patch for this to the tpm2-tss. So it will be available in later versions I think.

Reply:
But yeah I think we need it.

Reply:
Fixed